src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -393,18 +423,19 @@ private void AddTypeIfNotRegistered(Type type, TypeCollectionContext context)
                 if (namedType == typeof(PageInfoType))
                 {
                     AddType(new PageInfoType(), context);
-                    return;
                 }
-                if (namedType.IsGenericType)
+                else if (namedType.IsGenericType && (namedType.ImplementsGenericType(typeof(EdgeType<>)) || namedType.ImplementsGenericType(typeof(ConnectionType<,>))))
```

Comment:
Perhaps these types should be taken out of here. I've been thinking about this for a long time. Not in thin PR though.

Reply:
It would be better if upon finding a generic type (e.g. `EdgeType<MyModel>`), it uses reflection to retrieve the generic version of the type (e.g. `EdgeType<>`) and then checks `_builtInCustomScalars` for the generic type.  If matched, it will use `Activator.CreateInstance` to create an instance.  Fine as-is though.

Reply:
Better to make new issue and link it to #1039